### DISCUSSION TITLE
Spinning up container to run a one time task

### DISCUSSION THREAD

**Discussion by chrispage1** [basecamp/kamal#1206] [Q&A] [answered]
Hi! I'm having a go at using kamal to run PHP Laravel deployments and so far so good. I've successfully got a deployment up and running that runs the web service, schedule and queue workers.

The only thing I would like to add is autonomous migrations. So with Laravel, after deployment I'd like to run `php artisan migrate --force` on one of the containers. I don't need the container to be long lived. Would this be possible to achieve via a hook?

**Comment by tuladhar** [comment, marked as answer]
I believe this is what you are looking for: [Run interactive commands over SSH](https://kamal-deploy.org/docs/commands/running-commands-on-servers/#run-interactive-commands-over-ssh)

```
kamal app exec -i "php artisan migrate --force"
```

**Reply by chrispage1** [reply]
Thank you - thats great - I was just wondering if there's a way I can do it automatically as part of the deploy process?

**Reply by tuladhar** [reply]
You can use [post-deploy](https://kamal-deploy.org/docs/hooks/post-deploy/) hook.

**Reply by chrispage1** [reply]
Thank you - I have gone with the below for my `post-deploy` hook file

```bash
#!/bin/sh

kamal app exec --primary --roles=web --no-interactive "php artisan migrate --force"
```
